test/Microsoft.AspNetCore.Mvc.IntegrationTests/TryUpdateModelIntegrationTest.cs

```diff
@@ -512,6 +513,73 @@ private class Person5
             Assert.Empty(modelState);
         }
 
+        private class Person7
+        {
+            public IEnumerable<Address> Address { get; } = new Address[] { };
```

Comment:
nit: Could make test a `[Theory]` with different unwritable `IEnumerable<Address>` implementations e.g. `null` and a `ReadOnlyCollection<Address>`.